src/file/file.model.ts

```diff
@@ -121,14 +125,20 @@ fileSchema.pre('save', async function (next: NextFunction) {
   const name = (<any>this).name;
   const parent = (<any>this).parent;
   const ownerID = (<any>this).ownerID;
+  const appID = (<any>this).appID;
 
-  const query: any = { name, parent };
+  const parentFile = await fileModel.findById(parent);
+
+  const query: any = { name, parent, appID };
   if (!parent) {
     query.ownerID = ownerID;
   }
 
   const existingFile = await fileModel.findOne(query);
-  if (existingFile && !existingFile.float) {
+
+  if (parentFile && parentFile.appID !== appID) {
+    next(new FileParentAppIDNotEqual());
```

Comment:
Note that this means that the main drive-client wont be able to create files inside other applications folders.

Reply:
That's OK for now